### PR TITLE
feat(#476): implement GetHeaderForHeight method

### LINKED ISSUE #476: [FEATURE] Wallet: GetHeaderForHeight

### PR DESCRIPTION
**Pull request by jakubmkowalski**
Closes #476 
This pull request implements the `GetHeaderForHeight` functionality in the wallet package, enabling retrieval of block headers by height. It introduces a mapping layer to convert internal header representations, robust error handling, and comprehensive tests to ensure correct behavior and validation.

**Implementation of GetHeaderForHeight:**

- Adds the `GetHeaderForHeight` method to the `Wallet` struct, including validation for wallet services and originator, integration with the service layer, and mapping of results to the SDK type.
- Introduces the `MapGetHeaderResults` function in `mapping_get_header_results.go` to convert `ChainBaseBlockHeader` to `sdk.GetHeaderResult`, with error handling for nil results and byte conversion failures.

**Testing and validation:**

- Adds `wallet_get_header_for_height_test.go` with tests for argument validation, missing services, and service error scenarios, ensuring robust coverage for the new method.

**!!!Bug fixes and edge case handling:!!!**

- Updates `ChainBaseBlockHeader.Bytes()` to handle empty `PreviousHash` fields by initializing a 32-byte array, preventing errors when the field is missing.

### CODE REVIEW

File: pkg/wallet/wallet_get_header_for_height_test.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+			ctx: context.Background(),
+			args: sdk.GetHeaderArgs{
+				Height: 100,
+			},
+			originator: "test-originator",
+		},
+		"height 100000 (medium height)": {
+			ctx: context.Background(),
+			args: sdk.GetHeaderArgs{
+				Height: 1000,
```
[the remark this answers is not in the document]
Comment (jakubmkowalski) [reply pkg/wallet/wallet_get_header_for_height_test.go]: fixed

File: pkg/wdk/services_chain_block_headers.go [hunk 2]
```diff
@@ -70,9 +70,20 @@ func (c *ChainBaseBlockHeader) Hex() (string, error) {
...
+		hash = make([]byte, 32)
+	} else {
+		hash, err = hex.DecodeString(c.PreviousHash)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert 'previous hash' field into bytes slice: %w", err)
+		}
+
+		if len(hash) != 32 {
+			return nil, fmt.Errorf("'previous hash' field should be a 32 byte-hex length")
+		}
```
[the remark this answers is not in the document]
Comment (jakubmkowalski) [reply pkg/wdk/services_chain_block_headers.go:88]: removed redundant check

File: pkg/wdk/services_chain_block_headers.go [hunk 3]
```diff
@@ -70,13 +70,20 @@ func (c *ChainBaseBlockHeader) Hex() (string, error) {
...
-	hash, err := hex.DecodeString(c.PreviousHash)
-	if err != nil {
-		return nil, fmt.Errorf("failed to convert 'previous hash' field into bytes slice: %w", err)
-	}
+	var hash []byte
+	var err error
+
+	if c.PreviousHash == "" {
+		hash = make([]byte, 32)
+	} else {
```
Comment (chris-4chain) [inline pkg/wdk/services_chain_block_headers.go]: I assume the `if` is per "genesic block" - maybe, for better readability:
```suggestion
	genesis := c.PreviousHash == ""
	if genesis {
		hash = make([]byte, 32)
	} else {
```

File: pkg/wdk/services_chain_block_headers.go [hunk 4]
```diff
@@ -70,13 +70,20 @@ func (c *ChainBaseBlockHeader) Hex() (string, error) {
 // followed by time, bits, and nonce fields written in little-endian order.
 // Returns an error if any of the fields cannot be parsed or written.
 func (c *ChainBaseBlockHeader) Bytes() ([]byte, error) {
-	hash, err := hex.DecodeString(c.PreviousHash)
-	if err != nil {
-		return nil, fmt.Errorf("failed to convert 'previous hash' field into bytes slice: %w", err)
-	}
+	var hash []byte
```
Comment (chris-4chain) [inline pkg/wdk/services_chain_block_headers.go]: I would rename this to `prevHash`

**Review by chris-4chain — APPROVED** [review]

File: pkg/wdk/services_chain_block_headers.go (hunk 3 above)
Comment (jakubmkowalski) [reply pkg/wdk/services_chain_block_headers.go]: changed

File: pkg/wdk/services_chain_block_headers.go (hunk 4 above)
Comment (jakubmkowalski) [reply pkg/wdk/services_chain_block_headers.go]: renamed

**Review by cain4chain — APPROVED** [review]

**Review by chris-4chain — APPROVED** [review]

**Review by mgosek-4chain — APPROVED** [review]
